zarr/tests/test_dim_separator.py

```diff
@@ -63,6 +63,7 @@ def test_open(dataset):
     verify(zarr.open(dataset))
```

Comment:
I think there's [an additional problem](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=366681&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d&l=2822) with `test_open[static_flat]` and `test_open[static_nested]`, which is also blocking the Conda release.  The Pytest working directory does not contain the expected paths `fixture/flat` and `fixture/nested`, which becomes clearer if you're stricter about the persistence mode you use in `test_open`:
```suggestion
    verify(zarr.open(dataset, "r"))
```

Reply:
This has the added benefit of guarding against accidentally leaving any little droppings in the Pytest working directory.

Reply:
Hi @benjaminhwilliams, I at least welcome it! ;) Would you be up for opening this as a PR?

Reply:
Sure.  This small suggestion only highlights the issue, it doesn't fix the problem that the Pytest working directory is not the project root, as the test appears to assume.  But we can fix that by discussion in the PR.  I'll open it and request your review.